device_authorize_writer.go

```diff
@@ -0,0 +1,40 @@
+// Copyright © 2022 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright © 2015-2021 Aeneas Rekkas <aeneas+oss@aeneas.io>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author		Aeneas Rekkas <aeneas+oss@aeneas.io>
+ * @copyright 	2015-2021 Aeneas Rekkas <aeneas+oss@aeneas.io>
+ * @license 	Apache-2.0
+ *
+ */
+
+package fosite
+
+import (
+	"context"
+	"net/http"
+)
+
+// Once the user has approved the grant he will be redirected on his loggin machine
+// to a webpage (usally hosted in hydra-ui) to understand that he was connected successfully
+// and that he can close this tab and return to his non-interactive device;
+func (f *Fosite) WriteDeviceAuthorizeResponse(ctx context.Context, rw http.ResponseWriter, requester DeviceAuthorizeRequester, responder DeviceAuthorizeResponder) {
+	rw.Header().Set("Cache-Control", "no-store")
+	rw.Header().Set("Pragma", "no-cache")
+	rw.Header().Set("Location", f.Config.GetDeviceDone(ctx))
+	rw.WriteHeader(http.StatusSeeOther)
```

Comment:
Use `net/http.Redirect()` instead

Reply:
I've fixed this one but it require passing the http.Request onto the WriteDeviceAuthorizeResponse:
https://github.com/ory/fosite/blob/3313b71a11a71b2d75f2c18a9b31e41bd8a4d5d3/oauth2.go#L118

It defer from the Write*Response (witch don't have the http.Request as parameter):
https://github.com/ory/fosite/blob/3313b71a11a71b2d75f2c18a9b31e41bd8a4d5d3/oauth2.go#L104

If you prefer that I don't pass the http.Request, I can revert this particular commit.

Reply:
Also most of the writer use Raw Headers to perform redirections:

https://github.com/ory/fosite/blob/7efa846d221578f8716c7610632501631db8d27e/authorize_write.go#L64-L67